backend/backend.c

```diff
@@ -70,30 +72,31 @@ static struct wlr_backend *attempt_wl_backend(struct wl_display *display) {
 }
 
 struct wlr_backend *wlr_backend_autocreate(struct wl_display *display) {
-	struct wlr_backend *backend;
+	struct wlr_backend *backend = wlr_multi_backend_create(display);
+	if (!backend) {
+		wlr_log(L_ERROR, "could not allocate multibackend");
+		return NULL;
+	}
+
 	if (getenv("WAYLAND_DISPLAY") || getenv("_WAYLAND_DISPLAY")) {
-		backend = attempt_wl_backend(display);
-		if (backend) {
-			return backend;
+		struct wlr_backend *wl_backend = attempt_wl_backend(display);
+		if (wl_backend) {
+			wlr_multi_backend_add(backend, wl_backend);
 		}
 	}
 
 	const char *x11_display = getenv("DISPLAY");
 	if (x11_display) {
-		return wlr_x11_backend_create(display, x11_display);
+		struct wlr_backend *x11_backend =
+			wlr_x11_backend_create(display, x11_display);
+		wlr_multi_backend_add(backend, x11_backend);
```

Comment:
Yeah, the way this works is if you do `DISPLAY=:1 WAYLAND_DISPLAY=wayland-1 rootston` on a tty, it will open all three backends at the same time. Do you not like that?

Reply:
I don't want the X11 backend if I'm running Wayland (in GNOME, for instance).